flight/Modules/ManualControl/transmitter_control.c

```diff
@@ -306,8 +307,8 @@ int32_t transmitter_control_update()
 	    validChannel[MANUALCONTROLSETTINGS_CHANNELGROUPS_FLIGHTMODE];
 
 	// because arming is optional we must determine if it is needed before checking it is valid
-	bool arming_valid_input = settings.ChannelGroups[MANUALCONTROLSETTINGS_CHANNELGROUPS_ARMING] >= MANUALCONTROLSETTINGS_CHANNELGROUPS_NONE ||
-	    validChannel[MANUALCONTROLSETTINGS_CHANNELGROUPS_ARMING];
+	bool arming_valid_input = (settings.Arming >= MANUALCONTROLSETTINGS_ARMING_SWITCH) ||
```

Comment:
I put a comment in the XML file.  We have similar vulnerability to "NONE" in the channel mapping if we're concerned about that.  

I could add the 4 switch settings explicitly but it has a different problem if anyone ever adds another switch setting...

We really need to heavily refactor and clean this up in general.  Remember when we talked about it on IRC and you highlighted how mature it was in contrast to me saying the arming and failsafe logic is scary and disorganized?